# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/rustc/hotfix-elf-mips64n32.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- rustc-1.73.0-src/compiler/rustc_codegen_ssa/src/back/metadata.rs.vanilla	2023-10-12 19:11:54.409014170 +0200
+++ rustc-1.73.0-src/compiler/rustc_codegen_ssa/src/back/metadata.rs	2023-10-12 19:13:17.268016343 +0200
@@ -244,12 +244,11 @@
 
             let mut e_flags = elf::EF_MIPS_CPIC | arch;
 
-            // If the ABI is explicitly given, use it or default to O32.
-            match sess.target.options.llvm_abiname.to_lowercase().as_str() {
-                "n32" => e_flags |= elf::EF_MIPS_ABI2,
-                "o32" => e_flags |= elf::EF_MIPS_ABI_O32,
-                _ => e_flags |= elf::EF_MIPS_ABI_O32,
-            };
+            if sess.target.options.abi.contains("n32") {
+                e_flags |= elf::EF_MIPS_ABI2;
+            } else {
+                e_flags |= elf::EF_MIPS_ABI_O32;
+            }
 
             if sess.target.options.relocation_model != RelocModel::Static {
                 e_flags |= elf::EF_MIPS_PIC;
